lib/state/ui/middleware.ts

```diff
@@ -39,7 +48,46 @@ export default store => {
         }
         break;
     }
-
+    switch (action.type) {
+      case 'App.closeNote':
+      case 'App.showAllNotes':
+      case 'App.selectTrash':
+      case 'App.selectTag':
+        store.dispatch(setSelectedNote(null));
```

Comment:
these right here need not be in middleware. they can be left in the reducer to keep the complexity out of the system.

```ts
const setSelectedNote = (
   state = defaultNote,
   { type, note }: { type: string; note: T.NoteEntity[] }
 ) => {
  switch (type ) {
    case 'App.closeNote':
    case 'App.showAllNotes':
    case 'App.selectTrash':
    case 'App.selectTag':
      return null;
    case SET_SELECTED_NOTE:
      return note;
    default:
      return state;
  }
 };
```

Reply:
done